<launch>
    
    <arg name="odom_topic" default="/vins_simulator/noise/odom"/>
    <arg name="imu_topic" default="/vins_simulator/noise/imu"/>
    <arg name="vel_topic" default="/vins_simulator/noise/vel"/>
    <arg name="g_scale" default="1"/>

    <node pkg="eskf_fusion" type="run_eskf_lio" name="eskf_fusion" output="screen">
        <param name="imu_topic" value="$(arg imu_topic)" />
        <param name="odom_topic" value="$(arg odom_topic)" />
        <param name="vel_topic" value="$(arg vel_topic)" />

        <param name="lvio_pos_noise" type="double" value="0.5" />
        <param name="lvio_angle_noise" type="double" value="0.01" />

        <param name="acce_var" type="double" value="0.2" />
        <param name="gyro_var" type="double" value="0.05" />
        <param name="bias_acce_var" type="double" value="0.05" />
        <param name="bias_gyro_var" type="double" value="0.005" />

        <param name="vel_xy_noise" type="double" value="0.01" />
        <param name="vel_z_noise" type="double"  value="0.01" />

        <param name="g_scale" type="double" value="$(arg g_scale)" />
        
        <param name="update_bias_acc" type="bool" value="true" />
        <param name="update_bias_gyr" type="bool" value="true" />

    </node>

</launch>
